tracing-subscriber/src/fmt/time.rs

```diff
@@ -268,3 +268,210 @@ where
     timer.format_time(writer)?;
     write!(writer, " ")
 }
+
+/// A date/time type which exists primarily to convert `SystemTime` timestamps into an ISO 8601
+/// formatted string.
+///
+/// Yes, this exists. Before you have a heart attack, understand that the meat of this is musl's
+/// [`__secs_to_tm`][1] converted to Rust via [c2rust][2] and then cleaned up by hand. All existing
+/// `strftime`-like APIs I found were unable to handle the full range of timestamps representable
+/// by `SystemTime`, including `strftime` itself, since tm.tm_year is an int.
+///
+/// TODO: figure out how to properly attribute the MIT licensed musl project.
+///
+/// [1] http://git.musl-libc.org/cgit/musl/tree/src/time/__secs_to_tm.c
+/// [2] https://c2rust.com/
+///
+/// This is directly copy-pasted from https://github.com/danburkert/kudu-rs/blob/c9660067e5f4c1a54143f169b5eeb49446f82e54/src/timestamp.rs#L5-L18
```

Comment:
I directly copy pasted this code from along with tests, Is this the right way to give credits? or is there a better way? 
 @hawkw

Reply:
This personally works for me, although I also think it'd be fair to just credit musl, since the code is really just a machine translation of `__secs_to_tm` with very light cleanup (essentially doing whatever clippy said).